### DISCUSSION TITLE
Integration With Web-Based GUI Interface?

### DISCUSSION THREAD

**Discussion by nadolsw** [llmware-ai/llmware#269] [Ideas]
First of all - thank you so much for putting this repo together and maintaining it! 
It has been a huge help in getting personal experience with LLM and RAG systems.

IMO the one area that is currently lacking is some sort of tutorial or out-of-the-box support for configuring a front-end, locally-hosted,  web-based GUI (something like [Gradio](https://github.com/gradio-app/gradio) or [Label-Studio](https://github.com/HumanSignal/label-studio)) for the purpose of passing queries and responses to and from users.

Any resources you could compile or direct me to would be much appreciated!

**Comment by NYDocutest** [comment]
Hi Thank you for your kind feedback and for your question. We are looking into ways to provide web-based GUI support and are seeking help from the OS community first and hopefully we can find someone who is already working on this or may want to work on this as a first project. If we don't find help in the community, we will add this to our roadmap. Will keep you posted!

**Reply by xocolatl-aficionado** [reply]
@NYDocutest I have a PR ready #300 for review and testing. I could not run the test script on GitPod since it requires db connections to work (which wont on cloud dev envs i guess). But since I have made an entirely new file, am confident it hasnt broken anything.
